packages/nx/src/lock-file/npm-parser.ts

```diff
@@ -527,7 +528,13 @@ function nestMappedPackages(
   });
 
   if (initialSize === nestedNodes.size) {
-    throw Error('Loop detected while pruning. Please report this issue.');
+    output.error({
```

Comment:
Why is this reported so low? Shouldn't we have a `try catch` around `parseLockFile`?

Reply:
This error happens on `pruning` and although it should not fail anymore, we still want to show error log if it does so we can track it.

This is the only remaining place we anticipate that error might occur.

Reply:
I think we should catch even unanticipated errors.

Throwing an Error here is fine. But we should catch it higher up if the whole operation is not necessary (/ there is a workaround)

Reply:
This can be an error again right?

Reply:
The intention was to try not break the lock file creation, as even without a few packages missing the installation might still work.

But I guess it's safer to just return the root lock file.